rasa/nlu/components.py

```diff
@@ -102,6 +102,32 @@ def validate_only_one_tokenizer_is_used(pipeline: List["Component"]) -> None:
         )
 
 
+def validate_only_one_regex_component_is_used(pipeline: List["Component"]) -> None:
+    """Validates that RegexEntityExtractor and RegexFeaturizer are not used at the
+    same time.
+
+    Args:
+        pipeline: the list of the :class:`rasa.nlu.components.Component`.
+    """
+
+    from rasa.nlu.featurizers.sparse_featurizer.regex_featurizer import RegexFeaturizer
+    from rasa.nlu.extractors.regex_entity_extractor import RegexEntityExtractor
+
+    regex_featurizer_used = False
+    regex_entity_extractor_used = False
+    for component in pipeline:
+        if isinstance(component, RegexFeaturizer):
+            regex_featurizer_used = True
+        elif isinstance(component, RegexEntityExtractor):
+            regex_entity_extractor_used = True
+
+    if regex_featurizer_used and regex_entity_extractor_used:
+        raise InvalidConfigError(
```

Comment:
I would not raise here, but rather make a debug statement. I could imagine a situation where it useful to have both

Reply:
Just had a brief discussion with @koaning about it. Don't you think it might be confusing for users as the lookup tables/regexes are used differently/have a different meaning depending on the component? Also in the docs we point our that you can also use regexes for intent classification to indicate an utterance follows a certain schema and should most likely be classified as intent xyz. Those regexes should not be there if you use the `RegexEntityExtractor`.

Reply:
Not sure what you mean. Currently, `RegexFeaturizer` produces sentence and sequence features because of this. Do you suggest to remove the sentence features and update the docs to just mention that those features are good for entities?

Reply:
No, I meant in this case we only sequence regexes for entities

Reply:
then intent regexes will not influence `RegexEntityExtractor`

Reply:
Currently we cannot distinguish between intent and entity regexes. But maybe I guess we could filter out only those regexes in the `RegexEntityExtractor` which have a name that is in the list of entities. Will think about something.